### PR TITLE
Iasc 712 - upgrade to D9

### PR DESCRIPTION
**Pull request by attiks**

### CODE REVIEW

File: scripts/composer/ScriptHandler.php [hunk 1]
```diff
@@ -40,11 +40,9 @@ public static function createRequiredFiles(Event $event) {
       $fs->copy($drupalRoot . '/sites/default/default.settings.php', $drupalRoot . '/sites/default/settings.php');
       require_once $drupalRoot . '/core/includes/bootstrap.inc';
       require_once $drupalRoot . '/core/includes/install.inc';
-      $settings['config_directories'] = [
-        CONFIG_SYNC_DIRECTORY => (object) [
-          'value' => Path::makeRelative($drupalFinder->getComposerRoot() . '/config/sync', $drupalRoot),
-          'required' => TRUE,
-        ],
+      $settings['config_directories'] = (object) [
```
Comment (lazysoundsystem) [inline scripts/composer/ScriptHandler.php:43]: I think this should be `$settings['config_sync_directory']` as per https://www.drupal.org/node/3018145 but I'm not sure (https://github.com/UN-OCHA/pocam8-site/pull/78#discussion_r677593127)

File: composer.json [hunk 2]
```diff
@@ -20,46 +20,39 @@
         }
     ],
     "require": {
-        "php": ">=7.2",
-        "composer/composer": "^1.10.22",
-        "composer/installers": "^1.7",
+        "php": ">=7.3",
```
Comment (lazysoundsystem) [inline composer.json:23]: On https://github.com/UN-OCHA/pocam8-site/pull/78, cafuego suggested moving to php7.4 at the same time.

File: html/modules/custom/iasc_hero/src/Form/HeroRevisionRevertForm.php [hunk 3]
```diff
@@ -136,7 +136,7 @@ public function submitForm(array &$form, FormStateInterface $form_state) {
   protected function prepareRevertedRevision(HeroInterface $revision, FormStateInterface $form_state) {
     $revision->setNewRevision();
     $revision->isDefaultRevision(TRUE);
-    $revision->setRevisionCreationTime(REQUEST_TIME);
+    $revision->setRevisionCreationTime(time());
```
Comment (lazysoundsystem) [inline html/modules/custom/iasc_hero/src/Form/HeroRevisionRevertForm.php:139]: A bit picky, but the deprecation notice suggests something more verbose: https://www.drupal.org/node/2785211 (and in a few other instances in this module)

**Review by lazysoundsystem — CHANGES_REQUESTED** [review]
Looks good, but should bump to php7.4 and probably to composer 2 too.

File: scripts/composer/ScriptHandler.php (hunk 1 above)
Comment (attiks) [reply scripts/composer/ScriptHandler.php]: Nice catch, fixed

File: html/modules/custom/iasc_hero/src/Form/HeroRevisionRevertForm.php (hunk 3 above)
Comment (attiks) [reply html/modules/custom/iasc_hero/src/Form/HeroRevisionRevertForm.php:139]: I know, but injecting a service to get a timestamp felt like a bit of overload

File: scripts/composer/ScriptHandler.php [hunk 4]
```diff
@@ -40,11 +40,9 @@ public static function createRequiredFiles(Event $event) {
       $fs->copy($drupalRoot . '/sites/default/default.settings.php', $drupalRoot . '/sites/default/settings.php');
       require_once $drupalRoot . '/core/includes/bootstrap.inc';
       require_once $drupalRoot . '/core/includes/install.inc';
-      $settings['config_directories'] = [
-        CONFIG_SYNC_DIRECTORY => (object) [
-          'value' => Path::makeRelative($drupalFinder->getComposerRoot() . '/config/sync', $drupalRoot),
-          'required' => TRUE,
-        ],
+      $settings['config_sync_directory'] = (object) [
```
Comment (lazysoundsystem) [inline scripts/composer/ScriptHandler.php:43]: Not a blocker, but this section can be removed as it's overridden in the dockerfile - see https://github.com/UN-OCHA/slt8-site/pull/37#discussion_r678962876 (if it's needed for docksal development, let me know and I can replace it for SLT.)

File: composer.json [hunk 5]
```diff
@@ -20,47 +20,39 @@
         }
     ],
     "require": {
-        "php": ">=7.4",
-        "composer/composer": "^1.10.22",
-        "composer/installers": "^1.7",
-        "cweagans/composer-patches": "^1.7",
-        "drupal-composer/preserve-paths": "^0.1.6",
+        "php": ">=8.0",
+        "cweagans/composer-patches": "^1.6.5",
```
Comment (lazysoundsystem) [inline composer.json:24]: Don't think this needs to go down from ^1.7

**Review by lazysoundsystem — APPROVED** [review]
All looking good.

**Pull request merged**